test/test_linalg.py

```diff
@@ -2382,6 +2382,18 @@ def run_test_singular_input(batch_dim, n):
         for params in [(1, 0), (2, 0), (2, 1), (4, 0), (4, 2), (10, 2)]:
             run_test_singular_input(*params)
 
+    @skipCUDAIfNoMagmaAndNoCusolver
+    @skipCPUIfNoLapack
+    @onlyOnCPUAndCUDA   # TODO: XLA doesn't raise exception
+    @skipCUDAIfRocm
+    @dtypes(torch.float32, torch.float64, torch.complex64, torch.complex128)
+    def test_inverse_errors_large(self, device, dtype):
+        x = torch.empty((8, 10, 616, 616), dtype=dtype, device=device)
```

Comment:
Just so I understand correctly, this is creating a non-invertible matrix to trigger the problem that was causing the segfault correct?

Reply:
Yes, it's the same shape as the tensor in the issue but contains a simpler non-invertible matrix (identity with one diagonal value set to 0). Without filling `ipiv_data`, this test fails with the same memory access violation as the original issue.